### DISCUSSION TITLE
Make the theme styling direction agnostic

### DISCUSSION THREAD

**Discussion by yousef8** [satnaing/astro-paper#433] [Ideas]
Hi all,

I'm a native arabic speaker and i recently forked astro-paper theme to make my blog which will be in arabic.

To make this transition for the theme to be compatible with arabic which is a **RTL** (right to left) lang, first step was to edit the tailwind styling to be direction agnostic which mean the theme layout direction can be **RTL** or **LTR** without messing the styling this was relatively easy thanks to the css logical properties that helps to make the UI layout direction agnostic through logical properties ` margin-inline-start` and ` margin-inline-end` and this was first adopted in tailwind in version 3.3 , here is the  [PR that first introduced this feature to tailwind](https://github.com/tailwindlabs/tailwindcss/pull/10166) for more info.

After i finished this step, I figured that i can submit a PR to add this feature to the main theme which can be useful to anyone in the future that would want to utilize the theme for any other RTL language , and to save them a step they would have to do.

I know this can add an extra thing to keep in mind when mainting the theme, but i can offer that i will maintain this feature as i already will be syncing my fork with yours and if there is any bug i will issue a fix for it.

[Here is my current branch with the proposed feature](https://github.com/yousef8/ibn-elsayed-blog/tree/feat/direction-agnostic) you can inspect the commits and you will see the changes is very minimal.

**Comment by eerison** [comment]
I guess you could covert this discussion in an issue, I guess it gonna be a nice feature.

**Reply by yousef8** [reply]
great, i will do that

**Reply by eerison** [reply]
is it duplicated?
https://github.com/satnaing/astro-paper/issues/439

**Reply by yousef8** [reply]
indeed that is exactly what i wanted

**Comment by satnaing** [comment]
@yousef8 
I just made a PR about this issue_ RTL support. https://github.com/satnaing/astro-paper/pull/531
If you find some problems with my implementation, please let me know.

**Reply by yousef8** [reply]
sure thing, i will take a look at it
